internal/exec/selected/selected.go

```diff
@@ -89,7 +90,7 @@ func applySelectionSet(r *Request, s *resolvable.Schema, e *resolvable.Object, s
 				}
 
 			case "__schema":
-				if !r.DisableIntrospection {
+				if !r.DisableIntrospection && !r.DisableSchemaIntrospection {
```

Comment:
```
					if !r.DisableIntrospection || !r.DisableSchemaIntrospection {
```

Shouldn't this be `||` and not `&amp;&amp;`?

Reply:
No, it should be `&&`, but it took me a while to work it out too.  It gave me motivation to get the tests plumbed through to make sure.  :)

I've rewritten it via deMorgans law to the possibly easier-to-read variant that uses `||`:
```
if !(r.DisableIntrospection || r.DisableSchemaInspection) {
```

Basically we only want to append the fields if both `Disable*` options are false.